### PR TITLE
Extract api for UI from api v1 routes and remove session dependent for API/v1

### PR DESCRIPTION
**Pull request by lunny**
Currently, API/v1 depends on `Session` auth check but it should be removed. Because mixed UI API requests and API v1 will make something complicated.

There are many differences between the two APIs.

- UI API in fact should be consumed only by Gitea UI and could not be stable, it could be changed according to UI need and don't care about the break.
But API v1 will be exposed to external applications and should be stable and less break.

- UI API should read cookie and session like other routes but API v1 should only read HTTP header for the token when authentication, cookie and session is unnecessary.

- UI API should have csrf check but API v1 should not have csrf check.

- UI API should not support basic auth but API v1 should support basic auth and token auth

Currently, I copied the routes UI needed from v1 to UI but still use original functions. This is only a start and I think we can do more.

After this merged, API could be disabled or UI could be disabled if someone need that.

depends on #16455

### CODE REVIEW

**Comment by lunny** [comment]
depends on #16086

**Comment by lafriks** [comment]
tests are failing

File: modules/session/store.go [hunk 1]
```diff
@@ -10,3 +10,26 @@ type Store interface {
 	Set(interface{}, interface{}) error
 	Delete(interface{}) error
 }
+
+// EmptyStore represents an empty store
+type EmptyStore struct{}
```
Comment (6543) [inline modules/session/store.go]: It's not used everywhere - is it for some new thing in the future?

**Comment by 6543** [comment]
:eyes:

**Comment by 6543** [comment]
please resolve conflicts :eye:

**Pull request closed without merging**